Source/ASImageNode.mm

```diff
@@ -610,7 +610,14 @@ - (void)setNeedsDisplayWithCompletion:(void (^ _Nullable)(BOOL canceled))display
 - (void)tintColorDidChange
 {
   [super tintColorDidChange];
-  if (_image.renderingMode == UIImageRenderingModeAlwaysTemplate) {
+
+  BOOL isTemplateImage = NO;
+  {
+    AS::MutexLocker l(__instanceLock__);
+    isTemplateImage = (_image.renderingMode == UIImageRenderingModeAlwaysTemplate);
```

Comment:
Should we cache this information until `self.image` is assigned a new value? `UIImage` is immutable so this value shouldn't change unless we get a new image

Reply:
As discussed, nothing else in `ASImageNode` yet caches `UIImage` metadata, so I'd prefer not to have this diff be the first to do so.